File: java/string-2/endOther.java
/* Given two strings, return true if either of the strings appears at the 
 * very end of the other string, ignoring upper/lower case differences (in 
 * other words, the computation should not be "case sensitive").
 */
public boolean endOther(String a, String b) {
    if(a.length() < b.length()) {
        String temp = a;
        a = b.toLowerCase();
        b = temp.toLowerCase();
    }
                  
    return a.substring(a.length() - b.length()).equals(b);
}

File: python/string-2/end_other.py
# Given two strings, return True if either of the strings appears at the very 
# end of the other string, ignoring upper/lower case differences (in other 
# words, the computation should not be "case sensitive").
def end_other(a, b):
    a = a.lower()
    b = b.lower()
        
    return a.endswith(b) or b.endswith(a)
